File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/BehaviourExecution.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * Represents behaviour execution information
 */
class BehaviourExecution: Type() {

    fun processInstanceId() { 
        add {"processInstanceId" } 
    }

    fun identityId() { 
        add {"identityId" } 
    }

    fun state() { 
        add {"state" } 
    }

    fun name() { 
        add {"name" } 
    }

    fun duration() { 
        add {"duration" } 
    }

    fun behaviourId() { 
        add {"behaviourId" } 
    }

    fun created() { 
        add {"created" } 
    }

    fun updated() { 
        add {"updated" } 
    }
}

File: gaia-sdk-javascript/src/graphql/response/type/BehaviourExecution.ts


import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Represents behaviour execution information
*/
export interface BehaviourExecution {
    processInstanceId?:Uuid, 
    identityId?:Uuid, 
    state?:string, 
    name?:string, 
    duration?:number, 
    behaviourId?:Uuid, 
    created?:ISO8601, 
    updated?:ISO8601
}